### PR TITLE
prompt user on init if project has DESCRIPTION file

### PR DESCRIPTION
**Pull request by kevinushey**
Closes https://github.com/rstudio/renv/issues/1485.

### CODE REVIEW

**Comment by kevinushey** [comment]
Here's an example of the output.

```
> renv::init()
This project contains a DESCRIPTION file.
DESCRIPTION files are often used to explicitly declare a project's R dependencies.
How would you like to declare R package dependencies in this project?

1: I will record my project's package dependencies in the DESCRIPTION file.     (explicit mode)
2: I will let renv automatically determine the project's package dependencies.  (implicit mode)

Selection: 1
- Using 'explicit' snapshot type. Please see `?renv::snapshot` for more details.
```

One other question to ask: should we perform this check if we're calling `renv::init()` in a project that has already been initialized? (In other words, if `settings$snapshot.type()` has already been set?)

File: R/init.R [hunk 1]
```diff
@@ -307,3 +310,34 @@ renv_init_repos <- function() {
...
+
+    title = c(
+      "This project contains a DESCRIPTION file.",
+      "DESCRIPTION files are often used to explicitly declare a project's R dependencies.",
+      "How would you like to declare R package dependencies in this project?"
+    ),
+
+    choices = c(
+      explicit = "I will record my project's package dependencies in the DESCRIPTION file.     (explicit mode)",
+      implicit = "I will let renv automatically determine the project's package dependencies.  (implicit mode)"
```
Comment (hadley) [inline R/init.R:332]:
```suggestion
      explicit = "Take project dependencies only from the DESCRIPTION (explicit mode)",
      implicit = "Use all files to determine the dependencies (implicit mode)"
```

That's probably a bit too short, but in particularly the "I will" reads weirdly to me.

Comment (kevinushey) [reply R/init.R:332]: Thanks! This seems like an improvement to me.

The main thing I was trying to decide was whether we need to be clear that it's the user's responsibility to update the DESCRIPTION file, but maybe that's going to be obvious enough for anyone who has taken the time to actually create one?

Comment (hadley) [reply R/init.R]: Yeah, I think it's obvious enough from the context + message.

File: NEWS.md [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
   
 * `renv::restore()` now emits an error if called in a project that
   does not contain a lockfile. (#1474)
 
-* If `config$user.profile()` is `TRUE`, the packages it uses are now included
+* If `renv::config$user.profile()` is `TRUE`, the packages it uses are now included
   in the development dependencies, which means that they will be installed 
   by `install()` but not recorded in the snapshot.
 
 * `renv` now attempts to infer remote dependencies for GitHub packages that
```
Comment (hadley) [inline NEWS.md:22]: If you're touching this, you could do `` `renv` `` to `renv` too

File: tests/testthat/_snaps/init.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# init() prompts the user for the snapshot type
+
+    Code
+      init()
+    Output
+      This project contains a DESCRIPTION file.
```
Comment (hadley) [inline tests/testthat/_snaps/init.md:6]: I really like this new wording 😄

File: tests/testthat/_snaps/init.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      Which files should renv use for dependency discovery in this project?
+      
+      1: Use only the DESCRIPTION file. (explicit mode)
+      2: Use all files in this project. (implicit mode)
```
Comment (hadley) [inline tests/testthat/_snaps/init.md:10]: The holy grail: perfect alignment without extra spaces 😆

File: tests/testthat/test-init.R [hunk 5]
```diff
@@ -217,3 +217,13 @@ test_that("init() uses PPM by default", {
   expect_equal(repos, "https://packagemanager.posit.co/cran/latest")
 
 })
+
+test_that("init() prompts the user for the snapshot type", {
+  skip_on_cran()
+
+  project <- renv_tests_scope("bread")
+  writeLines("Depends: bread", con = "DESCRIPTION")
```
Comment (hadley) [inline tests/testthat/test-init.R:225]: Might be worth also installing bread just to keep the snapshot focussed?

File: tests/testthat/_snaps/init.md (hunk 4 above)
Comment (kevinushey) [reply tests/testthat/_snaps/init.md:10]: I tried very hard to make it happen 😄

**Pull request merged**